public/calculator.js

```diff
@@ -0,0 +1,145 @@
+				break;
+			case "-":
+				computation = prev - current;
+				break;
+			case "*":
+				computation = prev * current;
+				break;
+			case "÷":
+				computation = prev / current;
+				break;
+			default:
+				return;
+		}
+		this.currentOperand = computation;
+		this.operation = undefined;
+		this.previousOperand = "";
+		axios.post("/save-last-answer", {
+			lastAnswer: this.currentOperand
+		});
+		lastAnswerElement.innerText = this.currentOperand;
+	}
+	getDisplayNumber(number) {
+		const stringNumber = number.toString();
+		const integerDigits = parseFloat(stringNumber.split(".")[0]);
+		const decimalDigits = stringNumber.split(".")[1];
+		let integerDisplay;
+		if (isNaN(integerDigits)) {
+			integerDisplay = "";
+		} else {
+			integerDisplay = integerDigits.toLocaleString("en", {
+				maximumFractionDigits: 0,
+			});
+		}
+		if (decimalDigits != null) {
+			return `${integerDisplay}.${decimalDigits}
+    `;
+		} else {
+			return integerDisplay;
+		}
+	}
+	updateDisplay() {
+		this.currentOperandTextElement.innerText = this.getDisplayNumber(
+			this.currentOperand
+		);
+		if (this.operation != null) {
+			this.previousOperandTextElement.innerText = `${this.getDisplayNumber(
+				this.previousOperand
+			)}${this.operation}`;
+		} else {
+			this.previousOperandTextElement.innerText = "";
+		}
+	}
+}
+
+const numberButtons = document.querySelectorAll("[data-number]");
+const operationButtons = document.querySelectorAll("[data-operation]");
+const equalsButton = document.querySelector("[data-equals]");
+const deleteButton = document.querySelector("[data-delete]");
+const allClearButton = document.querySelector("[data-all-clear]");
+const previousOperandTextElement = document.querySelector(
+	"[data-previous-operand]"
+);
+const currentOperandTextElement = document.querySelector(
+	"[data-current-operand"
+);
+const lastAnswerElement = document.getElementById("lastAnswer");
+
+const calculator = new Calculator(
+	previousOperandTextElement,
+	currentOperandTextElement,
+	lastAnswerElement
+);
+
+numberButtons.forEach((button) => {
+	button.addEventListener("click", () => {
+		calculator.appendNumber(button.innerText);
+		calculator.updateDisplay();
+	});
+});
+operationButtons.forEach((button) => {
+	button.addEventListener("click", () => {
+		calculator.appendNumber(button.innerText);
+		calculator.updateDisplay();
+	});
+});
+
+operationButtons.forEach((button) => {
+	button.addEventListener("click", () => {
+		calculator.chooseOperation(button.innerText);
+		calculator.updateDisplay();
+	});
+});
+equalsButton.addEventListener("click", (button) => {
+	calculator.compute();
+	calculator.updateDisplay();
+});
+allClearButton.addEventListener("click", (button) => {
+	calculator.clear();
+	calculator.updateDisplay();
+});
+
+deleteButton.addEventListener("click", (button) => {
+	calculator.delete();
+	calculator.updateDisplay();
+});
+
+axios.get("retrieve-last-answer").then(({ data }) => {
+	lastAnswerElement.innerText = data;
+});
```

Comment:
When the script first runs we make a GET request to the server to retrieve the last answer from our `last-answer.dat` file. So even upon reloading the page it remembers the previous answer.